osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -382,11 +392,28 @@ private void updateRank()
                 return;
 
             ScoreRank newRank = RankFromScore(Accuracy.Value, ScoreResultCounts);
+            ScoreRank newMaxRank = RankFromScore(MaximumAccuracy.Value, ScoreResultCounts);
+            ScoreRank newMinRank = RankFromScore(MinimumAccuracy.Value, ScoreResultCounts);
+
+            // Check if the misses are >= 1, when the misses are more then 1, it clamp max rank no more then A.
+            // Check Accuracy.Value >= accuracy_cutoff_s is for the reduce unnecessary assign causing the icon flicking
+            ScoreResultCounts.TryGetValue(HitResult.Miss, out int missesCount);
+            if (missesCount > 1 && Accuracy.Value >= accuracy_cutoff_s)
+                newMaxRank = ScoreRank.A;
+            // detect is 100% acc, so it earn SS or SS+ rank
+            else if (Accuracy.Value == 1.0d)
+                newMaxRank = newRank;
```

Comment:
Why is all of this being done here?

Miss count bigger than 1 only prevents gaining S rank in osu! and taiko. catch and mania only look at the raw accuracy. So putting any ruleset-specific logic here is both unnecessary (because `RankFromScore` should already be handling this), and incorrect (because you're applying rules to rulesets that don't apply to them normally)

And what is that about "the icon flicking" too? The bindable debounces updates (by checking if the value actually changed). As long as you're not assigning to `{Minimum,Maximum}Rank.Value` every time, no "flicking" should be taking place.